friends/src/App.js

```diff
@@ -7,22 +7,27 @@ class App extends Component {
     super();
     this.state={
       friends:[],
-      friend: ''
-    };
-    const inputHandler = e => {
-      this.setState({ friend: e.target.value });
-    };
-    const addFriend = e => {
-      e.preventDefault();
-      const newFriend = {
-        name: this.state.friends.name,
-        age: this.state.friends.age,
-        email: this.state.friends.email
-      };
-      const friends = this.state.friends.slice();
-      friends.push(newFriend);
-      this.setState({ friend:'', friends });
-    }
+      friend: {
+        name: '',
+        age: '',
+        email: ''
+      }
+    }; 
+  }
+  inputHandler = e => {
+    this.setState({ friend: e.target.value });
```

Comment:
Here you need to set the state and change the correct property on `this.state.friend`

Try: 
```
this.setState({
  friend: { ...this.state.friend, [e.target.name]: e.target.value },
});
```